src/lib.rs

```diff
@@ -200,19 +200,15 @@ pub fn print_msg<P: AsRef<Path>>(
 pub fn handle_dump(meta: &Metadata, panic_info: &PanicInfo) -> Option<PathBuf> {
   let mut expl = String::new();
   #[cfg(feature = "nightly")]
-  let cause = match panic_info.message() {
-    Some(m) => format!("{}", m),
-    None => "Unknown".into(),
-  };
+  let message = panic_info.message();
 
   #[cfg(not(feature = "nightly"))]
-  let cause =
-    String::from("Error cause could not be determined by the application.");
+  let message = panic_info.payload().downcast_ref::<&str>();
```

Comment:
You should try downcasting to `String` too. `panic!("{}", 42);` will have a `String` as payload.

Reply:
Nice, thanks!